dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java

```diff
@@ -320,8 +320,11 @@ public Result queryProcessInstanceList(User loginUser, long projectCode, long pr
         }
 
         for (ProcessInstance processInstance : processInstances) {
-            processInstance.setDuration(
-                    DateUtils.format2Duration(processInstance.getStartTime(), processInstance.getEndTime()));
+            // if processInstance is running, the endTime should be the current time
+            String duration = processInstance.getState() != null && processInstance.getState().isFinished() ?
+                    DateUtils.format2Duration(processInstance.getStartTime(), processInstance.getEndTime()) :
+                    DateUtils.format2Duration(processInstance.getStartTime(), new Date());
```

Comment:
Can we move these code in a utils, `WorkflowUtils.getWorkflowInstanceDuration(ProcessInstance processInstance)`?

Reply:
good idea, done